autorest/azure/environments.go

```diff
@@ -101,6 +102,26 @@ var (
 	}
 )
 
+// EnvironmentFromName returns an Environment based on the common name specified
+func EnvironmentFromName(name string) (Environment, error) {
+	var env Environment
+	name = strings.ToUpper(name)
+	switch name {
```

Comment:
I prefer this to be just a simple map. Switch seems like an overkill here and makes the code longer.